extension/encoding/awscloudwatchmetricstreamsencodingextension/opentelemetry_unmarshaler.go

```diff
@@ -4,15 +4,62 @@
 package awscloudwatchmetricstreamsencodingextension // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/awscloudwatchmetricstreamsencodingextension"
 
 import (
+	"errors"
 	"fmt"
 
+	"github.com/gogo/protobuf/proto"
+	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/pmetric/pmetricotlp"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/awscloudwatchmetricstreamsencodingextension/internal/metadata"
+	expmetrics "github.com/open-telemetry/opentelemetry-collector-contrib/internal/exp/metrics"
 )
 
-type formatOpenTelemetry10Unmarshaler struct{}
+var errInvalidOTLPMessageLength = errors.New("unable to decode data length from message")
+
+type formatOpenTelemetry10Unmarshaler struct {
+	buildInfo component.BuildInfo
+}
 
 var _ pmetric.Unmarshaler = (*formatOpenTelemetry10Unmarshaler)(nil)
 
-func (f formatOpenTelemetry10Unmarshaler) UnmarshalMetrics(_ []byte) (pmetric.Metrics, error) {
-	return pmetric.Metrics{}, fmt.Errorf("UnmarshalMetrics unimplemented for format %q", formatOpenTelemetry10)
+func (f *formatOpenTelemetry10Unmarshaler) UnmarshalMetrics(record []byte) (pmetric.Metrics, error) {
+	md := pmetric.NewMetrics()
+	dataLen, pos := len(record), 0
+	for pos < dataLen {
+		// get start of the datum
+		n, nLen := proto.DecodeVarint(record)
+		if nLen == 0 && n == 0 {
+			return md, errInvalidOTLPMessageLength
+		}
+
+		// unmarshal datum
+		req := pmetricotlp.NewExportRequest()
+		pos += nLen
+		err := req.UnmarshalProto(record[pos : pos+int(n)])
+		pos += int(n)
+		if err != nil {
+			return pmetric.Metrics{}, fmt.Errorf("unable to unmarshal input: %w", err)
+		}
+
+		// add scope name and build info version to
+		// the resource metrics
+		for i := 0; i < req.Metrics().ResourceMetrics().Len(); i++ {
+			rm := req.Metrics().ResourceMetrics().At(i)
+			for j := 0; j < rm.ScopeMetrics().Len(); j++ {
+				sm := rm.ScopeMetrics().At(j)
+				sm.Scope().SetName(metadata.ScopeName)
+				sm.Scope().SetVersion(f.buildInfo.Version)
+			}
+		}
+
+		md = expmetrics.Merge(md, req.Metrics())
```

Comment:
Let's just stick to appending the resource metrics like in the Firehose receiver please.
`expmetrics.Merge` is expensive and it's not necessary for us to merge them in the unmarshaler.

Reply:
I think it's fine. As mentioned in the other thread, the JSON format is fundamentally different. If we didn't group there we would end up with a separate `plog.ResourceMetrics` for every data point, whereas the OTLP format already enables groupingby resources.